syntax = "proto3";

package transaction_verification;

service TransactionService {
    rpc verifyTransaction (CheckoutRequest) returns (Determination);
}

message CheckoutRequest {
    CreditCard creditcard = 1;
}

// A new message type for encapsulating credit card details.
message CreditCard {
    string number = 1; // Credit card number
    string expirationDate = 2; // Expiration date in MM/YYYY format
    string cvv = 3; // CVV code
}

message Determination {
    bool determination = 1;
}